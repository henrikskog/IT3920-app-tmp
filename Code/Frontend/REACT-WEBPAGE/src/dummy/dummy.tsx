import React, { useEffect, useState } from "react";

export function Test({ input }: { input?: number }) {
  const [blue, setBlue] = useState(0)

  useEffect(() => {
    if (input) {
      setBlue(input)
    }
  }, [input]);

  return (
    <>
      <h1>Hello World! {blue}</h1>
      <button
        onClick={() => {
          setBlue(blue - 1)
        }}
      >
        Press Me!
      </button>
    </>
  );
}
