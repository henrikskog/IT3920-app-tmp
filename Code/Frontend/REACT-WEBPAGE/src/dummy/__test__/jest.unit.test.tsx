import React from "react";
import { Test } from "../dummy";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import userEvent from '@testing-library/user-event'

describe("Jest tests", () => {
  it("Should match snapshot of test UI", () => {
    const component = renderer.create(<Test />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // "Should match snapshot of test UI with input"
    

  // Enzyme does not or will support React 18. Use RTL instead
  // https://stackoverflow.com/questions/72029189/does-enzyme-support-react-version-18
  // Migrate help: https://testing-library.com/docs/react-testing-library/migrate-from-enzyme#using-act-and-wrapperupdate 
  it("Should match input", () => {
    const value = 5;
    render(<Test input={value} />);

    expect(screen.getByRole("heading").innerHTML).toBe(`Hello World! ${value}`);
  });

  it("Should match input after interaction", async () => {
    const value = 3;
    render(<Test input={value} />);

    const user = userEvent.setup()

    await user.click(screen.getByRole("button"))

    expect(screen.getByRole("heading").innerHTML).toBe(`Hello World! ${value - 1}`);
  });
});
