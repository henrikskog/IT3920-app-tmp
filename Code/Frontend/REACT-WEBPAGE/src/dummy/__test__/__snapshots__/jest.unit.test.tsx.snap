// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest tests Should match snapshot of test UI 1`] = `
[
  <h1>
    Hello World! 
    0
  </h1>,
  <button
    onClick={[Function]}
  >
    Press Me!
  </button>,
]
`;
